### PR TITLE
Refactor GitHub action API

### LINKED ISSUE #35: Drop status input, use job.status instead
## Description

Whether the workflow has failed or not is already known through `job.status`.
We just need to read the info instead of having the user specifies it.

## Completion Criteria

- `status` input is removed
- GitHub action behaves similarly

## Implementation Notes / Suggestions

N/A

## Testing Notes / Suggestions

N/A

### PR DESCRIPTION
**Pull request by thomas-moulard**
Instead of hardcoding the metrics in the action, let the user decides what is needed.

Fix #35, #36

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -31,60 +35,55 @@ jobs:
...
-3. Send the status to CloudWatch Metrics.
+Defaults to:
 
-## Inputs
+```JSON
+    [
+      { 'Name': 'github.event_name', 'Value': "${{ github.event_name }}" },
+      { 'Name': 'github.ref', 'Value': "${{ github.ref }}" },
+      { 'Name': 'github.repository', 'Value': "${{ github.repository }}" },
+      { 'Name': 'github.workflow', 'Value': "${{ github.workflow }}" },
```
Comment (piraka9011) [inline README.md:65]: nit: Double quotes because it's JSON.

File: .prettierrc [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  "parser": "typescript",
+  "singleQuote": false,
+  "semi": true,
+  "tabWidth": 2,
+  "useTabs": true
```
Comment (piraka9011) [inline .prettierrc:6]: not spaces? :)

File: action.yml [hunk 3]
```diff
@@ -1,15 +1,28 @@
...
-  project-name:
-    description: 'Name of the project'
-    default: ${{ github.repository }}
+  metric-name:
+    description: 'Name of the metric to be published'
+    default: 'Builds'
+
+  metric-value:
+    description: 'Value of the metric to be published'
+    default: ${{ job.status	}} == 'success' ? 1.0 : 0.0
```
Comment (piraka9011) [inline action.yml:10]: extra spaces?

File: .prettierrc (hunk 2 above)
Comment (thomas-moulard) [reply .prettierrc:6]: This one is from setup-ros, so keeping for consistency, but agree with you in the principle.

**Comment by thomas-moulard** [comment]
@piraka9011 resolved all issues, trying to add the e2e test now.

**Pull request merged**